Solution1/ConsoleApp/Services/UserCommunicateService.cs

```diff
@@ -13,60 +21,123 @@ public class UserCommunicateService : IUserCommunicateService
-                return true;
+                case 0:
+                    _invoker.SetCommand(new CurrentlyWeatherCommand(this));
+                    break;
+                case 1:
+                    _invoker.SetCommand(new ForecastWeatherCommand(this));
+                    break;
+                case 2:
+                    _invoker.SetCommand(new ExitCommand());
+                    break;
             }
 
-            var weatherRepresentation = (string)default;
             try
             {
-                weatherRepresentation = (await _weatherServiсe.GetByCityNameAsync(cityName)).GetStringRepresentation();
+                await _invoker.RunAsync();
             }
             catch (HttpRequestException ex)
             {
                 if (ex.StatusCode == HttpStatusCode.NotFound)
                 {
                     Console.WriteLine(Constants.Errors.BadCityName);
-                    _logger.LogError($"{DateTime.Now}| Status code: {(int)HttpStatusCode.NotFound} {HttpStatusCode.NotFound}. User entered incorrect city name.");
-                    return true;
+                    _logger.LogError($"Status code: {(int)ex.StatusCode} {ex.StatusCode}. User entered incorrect city name.");
                 }
                 else
                 {
                     Console.WriteLine(Constants.Errors.RequestError);
-                    _logger.LogError($"{DateTime.Now}| Status code: {(int)ex.StatusCode} {ex.StatusCode}. {ex.Message}");
-                    return true;
+                    _logger.LogError($"Status code: {(int)ex.StatusCode} {ex.StatusCode}. {ex.Message}");
+                }
+            }
+            catch (ValidationException ex)
+            {
+                foreach (var error in ex.Errors)
+                {
+                    _logger.LogError(error.ErrorMessage);
+
+                    if (error.PropertyName == nameof(ForecastWeatherDTO.CityName))
+                    {
+                        Console.WriteLine(Constants.Validation.IncorrectCityName);
+                    }
+
+                    if (error.PropertyName == nameof(ForecastWeatherDTO.WeatherForPeriod))
+                    {
+                        Console.WriteLine(Constants.Validation.IncorrectPeriod);
+                    }
                 }
             }
             catch (Exception ex)
             {
                 Console.WriteLine(Constants.Errors.UnexpectedError);
-                _logger.LogError($"{DateTime.Now}| {ex.Message}");
-                return true;
+                _logger.LogError(ex.Message);
+            }
+        }
+
+        public async Task GetCurrentlyWeatherAsync()
+        {
+            var weatherRequest = new ForecastWeatherRequestDTO();
+
+            Console.WriteLine("Please, enter city name:");
+            weatherRequest.CityName = Console.ReadLine();
+
+            var weather = await _weatherServiсe.GetByCityNameAsync(weatherRequest);
```

Comment:
pass just name, form request where it's really necessary. obviously not here. i said just console should be left here and nothing else

Reply:
+